crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -187,31 +195,113 @@ impl C8YHttpProxyActor {
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
         let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let req_build_closure = |jwt_token: String, _internal_id: String| {
+            let request_internal_id = HttpRequestBuilder::get(url_get_id.clone());
+            request_internal_id.bearer_auth(jwt_token)
+        };
+
+        let token = if let Ok(token) = self.peers.jwt.await_response(()).await? {
+            self.end_point.token = token.clone();
+            token
+        } else {
+            return Err(C8YRestError::CustomError("JWT token not available".into()));
+        };
 
-        let request_internal_id = HttpRequestBuilder::get(url_get_id);
-        let res = self.execute(request_internal_id).await?;
+        let res = self
+            .execute_to_get_interal_id(
+                token,
+                self.end_point.c8y_internal_id.clone(),
```

Comment:
This kind of code makes the reader waste his time. As it's first surprising to pass an internal id to a method used to get the internal id. Then to realize that this internal id is not used at all but cloned many times.

=> Simply remove this argument

Reply:
done